xml5ever/src/tree_builder/mod.rs

```diff
@@ -692,7 +685,6 @@ impl<Handle, Sink> XmlTreeBuilder<Handle, Sink>
                     };
                     if tag.name.local == local_name!("script") {
                         self.insert_tag(tag.clone());
-                        self.complete_script();
```

Comment:
You need to return early when you encounter a script end tag. Same goes for the other `complete_script` call below.

Reply:
Will check next week how it can be implemented.